PixaView/Controller/PixaViewController.swift

```diff
@@ -0,0 +1,276 @@
+    //view Will Transition
+    //https://stackoverflow.com/questions/37152071/landscape-orientation-for-collection-view-in-swift
+    
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        
+        switch UIDevice.current.orientation {
+        case .landscapeLeft, .landscapeRight:
+            coordinator.animate (alongsideTransition: { (_) in
+                self.landscapeLayoutCollectionView()
+            }, completion: nil )
+        case .portrait, .portraitUpsideDown:
+            coordinator.animate (alongsideTransition: { (_) in
+                self.portaitCollectionView()
+            }, completion: nil )
+        default:
+            print("error")
+        }
+        super.viewWillTransition(to: size, with: coordinator)
+    }
+    
+    //https://stackoverflow.com/questions/38894031/swift-how-to-detect-orientation-changes
+    
+    // Portrait orientation
+    private func portaitCollectionView(){
+        let numberOfItemsPerRow : CGFloat = 1
+        let lineSpacing:CGFloat = 5
+        let interItemSpacing :CGFloat = 5
+        let width = ( collectionView.frame.width - CGFloat((numberOfItemsPerRow)) * interItemSpacing) / numberOfItemsPerRow
+        let height = width
+        
+        collectionViewFlowLayout .sectionInsetReference = .fromSafeArea
+        collectionViewFlowLayout.collectionView?.contentMode = .scaleAspectFit
+        collectionViewFlowLayout.itemSize = CGSize (width: width, height: height)
+        collectionViewFlowLayout.sectionInset = UIEdgeInsets.zero
+        collectionViewFlowLayout.scrollDirection = .vertical
+        collectionViewFlowLayout.minimumLineSpacing = lineSpacing
+        collectionViewFlowLayout.minimumInteritemSpacing = interItemSpacing
+        collectionView.setCollectionViewLayout(collectionViewFlowLayout, animated: true)
+    }
+    
+    // Landscape orientation
+    private func landscapeLayoutCollectionView(){
+       
+        let numberOfItemsInRow: CGFloat = 2
+        let lineSpacing: CGFloat = 2
+        let interItemSpacing: CGFloat = 2
+        
+        collectionViewFlowLayout.sectionInset = UIEdgeInsets.zero
+        collectionViewFlowLayout.scrollDirection = .vertical
+        collectionViewFlowLayout.minimumLineSpacing = lineSpacing
+        collectionViewFlowLayout .sectionInsetReference = .fromSafeArea
+        collectionViewFlowLayout.minimumInteritemSpacing = interItemSpacing
+        
+        let width = ( collectionView.frame.width - CGFloat((numberOfItemsInRow - 1)) * interItemSpacing) / numberOfItemsInRow
+        let height = width
+        
+        collectionViewFlowLayout.itemSize = CGSize(width: width, height: height)
+        collectionView.setCollectionViewLayout(collectionViewFlowLayout, animated: true)
+    }
+    
+    // didSelectItemAt indexPath
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
```

Comment:
Use Post model to but tags and image URL to DetailedViewController.

Reply:
need to discuss this idea